torch/_C/__init__.pyi.in

```diff
@@ -1839,7 +1839,8 @@ def _activate_cuda_trace() -> None: ...
 def _current_graph_task_id() -> _int: ...
 def _current_autograd_node() -> _Node: ...
 
-class _OutOfMemoryError: ...
+# Defined in torch/csrc/Exceptions.cpp
+class _OutOfMemoryError(RuntimeError): ...
```

Comment:
Rather annoying this doesn't inherit from MemoryError.

Reply:
Would it be good to change the Exceptions.cpp bindings to PyExc_MemoryError then? Or would that possibly cause breaking changes? I'm new to Python C api so I'm not sure.

Reply:
Might be a breaking change, but one that will not affect a lot of projects in practice. This PR fixes the immediate typing issue though so perhaps we can merge this and then open a new PR that does try to see if we can move it to a PyExc_MemoryError without breaking everything.